File: bench/bench.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"time"

	"github.com/Djarvur/parsewords"
)

const iterations = 100

func main() {
	bytes, err := ioutil.ReadAll(os.Stdin)
	if err != nil {
		panic(err)
	}

	text := string(bytes)

	var words []string

	started := time.Now()
	for ri := 0; ri < iterations; ri++ {
		words, err = parsewords.Shellwords(text)
		if err != nil {
			panic(err)
		}
	}
	spent := time.Since(started)

	fmt.Printf(
		"%d iterations done in %v: %d words found in %d bytes of input\n",
		iterations,
		spent,
		len(words),
		len(text),
	)
}

File: bench/bench.pl
use Text::ParseWords;
use Time::HiRes qw(time);

my $iterations = 100;                      

my $text = do { local $/; <STDIN> };

my @words = ();

my $started = time();
for (my $ri = 0; $ri < $iterations; $ri++) {
  @words = shellwords($text);
}
my $spent = time() - $started;

printf(
  "%d iterations done in %fs: %d words found in %d bytes of input\n",
  $iterations,
  $spent,
  scalar(@words), 
  length($text),
)